tests/compiler/test_typing.py

```diff
@@ -380,3 +380,241 @@ def test_2243_override_optional(snippetcompiler) -> None:
 Child()
         """,
     )
+
+
+def test_deprecate_number(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number val
+end
+        """,
+    )
+    with pytest.warns(
+        TypeDeprecationWarning,
+        match=re.escape("Type 'number' is deprecated, use 'float' or 'int' instead"),
+    ):
+        (_, scopes) = compiler.do_compile()
+
+
+def test_number_type(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    number i = 0
+end
+implement Test using std::none
+Test(i = 42)
+Test(i = -42)
+Test()
+Test(i = 42.0)
+Test(i = false)
+        """,
+    )
+    compiler.do_compile()
+
+
+def test_same_value_float_int(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+    i = 42.0
+    j = 42
+    i = j
+    j = i
```

Comment:
Can you also add a similar test verifying the `==` operator? i.e. `(42 == 42.0) == true`.